src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,33 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    static final String DATE_FORMAT = "dd.MM.yyyy";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter).minusDays(1);
+        LocalDate finishDate = LocalDate.parse(dateTo, formatter).plusDays(1);
+        StringBuilder builder = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+        for (String name : names) {
+            int salaryTotal = 0;
+            for (String datum : data) {
+                int salaryPerDay = Integer.parseInt(datum.split(" ")[2])
```

Comment:
```suggestion
            for (String datum : data) {
                String recordData = datum.split(" ");
                int salaryPerDay = Integer.parseInt(recordData[2])
```
так в тебе не буде багаторазово викликатись операція split і створюватись нові об'єкти